temporal-test-server/src/main/java/io/temporal/internal/testservice/StateMachines.java

```diff
@@ -789,7 +789,12 @@ private static void timeoutNexusOperation(
   private static State failNexusOperation(
       RequestContext ctx, NexusOperationData data, Failure failure, long notUsed) {
     RetryState retryState = attemptNexusOperationRetry(ctx, Optional.of(failure), data);
-    if (retryState == RetryState.RETRY_STATE_IN_PROGRESS) {
+    if (retryState == RetryState.RETRY_STATE_IN_PROGRESS
```

Comment:
Hm but in the test why are we even reaching this point since the error thrown should never fail the operation?

Reply:
This function is also for handling task failures. I guess maybe @bergundy was right and I should separate out a `FAIL_ATTEMPT` from failing the operation as a whole. But this is how workflow and activity task failures/retries are handled and I though it would be better to stay consistent and refactor them all at the same time.